packages/location-field/src/index.tsx

```diff
@@ -0,0 +1,758 @@
+      //    to 'search' geocoding
+      //  - If pressed when dropdown results menu is active, apply the location
+      //    associated with current selected menu item
+      case "Enter":
+        if (typeof activeIndex === "number") {
+          // Menu is active
+          // Retrieve location selection handler from lookup object and invoke
+          const locationSelected = locationSelectedLookup[activeIndex];
+          if (locationSelected) locationSelected();
+
+          // Clear selection & hide the menu
+          setMenuVisible(false);
+          setActiveIndex(null);
+        } else {
+          // Menu not active; get geocode 'search' results
+          geocodeSearch(evt.target.value);
+          // Ensure menu is visible.
+          setMenuVisible(true);
+        }
+
+        // Suppress default 'Enter' behavior which causes page to reload
+        evt.preventDefault();
+        break;
+      case "Escape":
+      case "Tab":
+        // Clear selection & hide the menu
+        setMenuVisible(false);
+        setActiveIndex(null);
+        break;
+      // Any other key pressed: clear active selection
+      default:
+        setActiveIndex(null);
+        break;
+    }
+  };
+
+  const geocodeSearch = text => {
+    if (!text) {
+      console.warn("No text entry provided for geocode search.");
+      return;
+    }
+    getGeocoder(geocoderConfig)
+      .search({ text })
+      .then(result => {
+        if (result.features && result.features.length > 0) {
+          // Only replace geocode items if results were found
+          setGeocodedFeatures(result.features);
+          setMessage(null);
+        } else {
+          console.warn(
+            "No results found for geocode search. Not replacing results."
+          );
+        }
+      })
+      .catch(err => {
+        console.error(err);
+      });
+  };
+
+  const renderFeature = (itemIndex, feature) => {
+    // generate the friendly labels for this feature
+    const { main, secondary } = generateLabel(feature.properties);
+
+    // Create the selection handler
+    const locationSelected = () => {
+      getGeocoder(geocoderConfig)
+        .getLocationFromGeocodedFeature(feature)
+        .then(geocodedLocation => {
+          // add the friendly location labels for use later on
+          geocodedLocation.main = main;
+          geocodedLocation.secondary = secondary;
+          geocodedLocation.name = getCombinedLabel(feature.properties);
+          // Set the current location
+          setLocation(geocodedLocation, "GEOCODE");
+          // Add to the location search history. This is intended to
+          // populate the sessionSearches array.
+          addLocationSearch({ location: geocodedLocation });
+        });
+    };
+
+    // Add to the selection handler lookup (for use in onKeyDown)
+    locationSelectedLookup[itemIndex] = locationSelected;
+
+    // Extract GTFS/POI info and assign to class
+    const { layer, source, id } = feature.properties;
```

Comment:
Sort props.

Reply:
Addressed in 18d5606109e7dfce7d00d42b42c8b1b6f075a0bc